autofac-tests/Ping/Providers/DatabasePingProvider.cs

```diff
@@ -0,0 +1,15 @@
+namespace autofac_tests
+{
+    using System;
+
+    public class DatabasePingProvider : IPingProvider
+    {
+        private static readonly Random rnd = new Random(Environment.TickCount);
+        public Func<Uri, bool> Ping { get; } = uri =>
+        {
+            var next = rnd.NextBool(40);
+            Console.WriteLine($"[DatabasePingProvider] - [{next}]");
```

Comment:
I would suggest to use abstract output like Log.Write and setup log4net (or any other tool) to log to console. Code will be more portable to different environments like services, workers e.t.c